### PR TITLE
use bounded wildcards for errorHandler (fixes #5045)

### LINKED ISSUE #5045: 2.x: RxJavaPlugin.get/setXXX generics?
For unit tests I frequently bump into the restrictiveness of 

```java
RxJavaPlugins.setErrorHandler(Consumer<Throwable> handler);
```

I would like the signature to be 

```java
RxJavaPlugins.setErrorHandler(Consumer<? super Throwable> handler);
```

There are many more methods in `RxJavaPlugins` that could support lower and upper bounded wildcards. Can I make these changes?

### PR DESCRIPTION
**Pull request by jschneider**
Adding bound wildcards for error handler - fixes #5045

### CODE REVIEW

**Comment by davidmoten** [comment]
yep that looks fine. I can do the others.

**Comment by davidmoten** [comment]
@akarnokd do you want test coverage of the signature changes?

**Comment by akarnokd** [comment]
Yes, make sure the contravariant consumer now compiles.

**Comment by davidmoten** [comment]
@jschneider can you add unit tests for this please?

**Comment by jschneider** [comment]
I will add a test. Yes.

**Comment by jschneider** [comment]
Added the test

File: src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java [hunk 1]
```diff
@@ -1168,7 +1168,28 @@ public void uncaughtException(Thread t, Throwable e) {
...
     }
 
-    @SuppressWarnings({ "rawtypes", "unchecked" })
+    /**
+     * Ensure setErrorHandler() accepts a consumer with "? super Throwable"
+     */
+    @Test
+    public void onErrorWithSuper() throws Exception {
+        try {
+            Consumer<? super Throwable> errorHandler = new Consumer<Throwable>() {
```
Comment (akarnokd) [inline src/test/java/io/reactivex/plugins/RxJavaPluginsTest.java:1177]: A better type would be `Consumer<Object> errorHandler = ...`

**Comment by akarnokd** [comment]
Looks like this got a conflict due to changes to RxJavaPlugins by other PRs. Could you rebase this onto the head?

**Comment by jschneider** [comment]
yep. I am on it

**Comment by jschneider** [comment]
should be good now.

**Review by akarnokd — APPROVED** [review]

**Pull request merged**